### PR TITLE
Add CI test for ceph, bug fix and cleanup

### PR DESCRIPTION
**Pull request by yangzhang-ibm**
1. Implementing the Ursula CI test with ceph;
2. Avoid compute hostname to be included in /etc/hosts of controller nodes, the issue was there before since controllers are also included in the [compute] host group.
3. Removing the unused playbook tasks which were part of old CI test framework: create.yml, delete.yml, keypair.yml. All these actions have been implemented with heat provisioner approach in new CI test framework.

### CODE REVIEW

**Comment by yangzhang-ibm** [comment]
@timrchavez, @j2sol, @notnownikki: could you help review and merge this PR, thanks!

**Comment by j2sol** [comment]
Wait a sec, does this mean that with our heat method, we don't have compute services on the controllers as well? (the intent is that they're converged controllers, so all three VMs should have compute services on them).

**Comment by yangzhang-ibm** [comment]
@j2sol No big change after this commit, I was just trying to remove a wrong entry in /etc/hosts files in controller nodes like following: 
```
127.0.1.1   ceph-swift-test-compute-0 ceph-swift-test-compute-0 ceph-swift-test-compute-0
```

That's because the code logic in pre-deploy.yml is as follows:
```    - name: add compute node to /etc/hosts
      lineinfile: dest=/etc/hosts regexp={{ testenv_instance_names[2] }}
                  insertafter=EOF
                  line="127.0.1.1 {{ testenv_instance_names[2] }}"
```

So that task is only needed for compute-0 node only.

**Comment by yangzhang-ibm** [comment]
This commit is just to fix that mistake during pre-deploy actions, but anyway the final version of /etc/hosts files on all nodes will be overwritten with the expected content during actual deployment with "ursula envs/test site.yml" which is the later step.

**Comment by notnownikki** [comment]
ok to test

**Comment by yangzhang-ibm** [comment]
The PR test failed due to neutron network issue in test env, it seems most recent PR tests were failing with that error.

**Comment by notnownikki** [comment]
retest

**Comment by notnownikki** [comment]
retest

**Comment by j2sol** [comment]
retest

**Comment by yangzhang-ibm** [comment]
Added another commit into this PR: Implement Ursula CI test with ceph

**Comment by timrchavez** [comment]
retest

File: playbooks/tests/tasks/ceph.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+- name: test various things from the ceph_osds
+  hosts: ceph_osds
+  tasks:
+  - name: check ceph cluster health
+    shell: ceph health
```
Comment (j2sol) [inline playbooks/tests/tasks/ceph.yml]: These should be command, not shell

File: playbooks/tests/tasks/ceph.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  - name: check ceph osd status
+    shell: ceph osd stat
```
Comment (j2sol) [inline playbooks/tests/tasks/ceph.yml]: command

File: playbooks/tests/tasks/ceph.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+- name: check ceph related configuration on controllers
+  hosts: controller
```
Comment (j2sol) [inline playbooks/tests/tasks/ceph.yml]: You can't define hosts in a task. hosts belongs to a playbook, not a task file.

File: playbooks/tests/tasks/ceph.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  tasks:
```
Comment (j2sol) [inline playbooks/tests/tasks/ceph.yml]: This is a tasks file, you can't have a tasks block.

File: playbooks/ci-ceph_swift/tasks/pre-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+- name: controller 1 hosts file
+  hosts: controller[1]
+  vars_files:
+  - ../vars/main.yml
+  tasks:
+    - name: add controller node to /etc/hosts
+      lineinfile: dest=/etc/hosts regexp={{ testenv_instance_names[1] }}
+                  insertafter=EOF
+                  line="127.0.1.1 {{ testenv_instance_names[1] }}"
```
Comment (j2sol) [inline playbooks/ci-ceph_swift/tasks/pre-deploy.yml]: Seems like these plays could be combined into one, with a loop.

File: playbooks/ci-ceph_swift/tasks/pre-deploy.yml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+- name: compute 0 hosts file
+  hosts: compute[0]
+  vars_files:
+  - ../vars/main.yml
+  tasks:
+    - name: add compute node to /etc/hosts
+      lineinfile: dest=/etc/hosts regexp={{ testenv_instance_names[2] }}
+                  insertafter=EOF
+                  line="127.0.1.1 {{ testenv_instance_names[2] }}"
```
Comment (j2sol) [inline playbooks/ci-ceph_swift/tasks/pre-deploy.yml]: These seem pretty static and hardcoded, and thus fragile if we wanted to expand the group members.

File: playbooks/ci-ceph_swift/tasks/pre-deploy.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+- name: umount ephemeral disk on ceph_osds
+  hosts: ceph_osds
+  tasks:
+    - name: umount ephemeral disk on ceph_osds
+      shell: umount /dev/{{ item }}
+      with_items: "{{ ceph.disks }}"
+
+- name: tasks for all
+  hosts: all
+  serial: 10
```
Comment (j2sol) [inline playbooks/ci-ceph_swift/tasks/pre-deploy.yml]: Why serial? And you're doing serial 10, which is more than the number of nodes anyway.

**Comment by j2sol** [comment]
Merging as is, can improve how the playbooks/tests/ structure looks as a follow up.

**Comment by yangzhang-ibm** [comment]
@j2sol Thanks for your review comments and merging this PR, I have taken a note of all the enhancements needed as you mentioned, and I will try to improve these items later.

**Pull request merged**